src/Compilers/CSharp/Portable/Binder/PatternExplainer.cs

```diff
@@ -96,6 +98,76 @@ int distance(BoundDecisionDagNode x)
             return result.ToImmutableAndFree();
         }
 
+        /// <summary>
+        /// Enumerates the paths from the root node to the node of interest, and invokes the handler
+        /// on each one until the handler returns false.
+        /// The order is deterministic, but we're not starting from the shortest path.
+        /// </summary>
+        /// <param name="rootNode">The root node of the DAG.</param>
+        /// <param name="targetNode">The node of interest.</param>
+        /// <param name="nullPaths">Whether to permit following paths that test for null.</param>
+        /// <param name="handler">Handler to call back for every path to the target node.</param>
+        private static void VisitPathsToNode(BoundDecisionDagNode rootNode, BoundDecisionDagNode targetNode, bool nullPaths,
+            Func<ImmutableArray<BoundDecisionDagNode>, bool, bool> handler)
+        {
+            var pathBuilder = new ArrayBuilder<BoundDecisionDagNode>();
+            exploreToNode(rootNode, currentRequiresFalseWhenClause: false);
+            return;
+
+            // Recursive exploration helper
+            // Returns true to continue, false to stop
+            bool exploreToNode(BoundDecisionDagNode currentNode, bool currentRequiresFalseWhenClause)
+            {
+                if (currentNode == targetNode)
+                {
+                    return handler(pathBuilder.ToImmutable(), currentRequiresFalseWhenClause);
+                }
+
+                pathBuilder.Push(currentNode);
+
+                switch (currentNode)
+                {
+                    case null:
```

Comment:
>case null [](http://example.com/codeflow?start=20&length=9)

Is `null` an expected case? If so, consider testing if it's obvious how to hit this case.

Reply:
DAG evaluation nodes only have a true branch for instance. Added a test.